File: Codeforces/C++/problemset/Spy Detected.cpp
// Problem Link: https://codeforces.com/problemset/problem/1512/A

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll n, t;
    cin >> t;
    for(ll j = 0; j < t; j++)
    {
        cin >> n;
        ll *a = new ll[n];
        for(ll i = 0; i < n; i++)
            cin >> a[i];
        for(ll i = 0; i < n; i++)
        {
            if(i == 0 && a[i] != a[i+1] && a[i] != a[i+2])
            {
                cout << i+1 << "\n";
                delete [] a;
                break;
            }
            else if(i > 0 && a[i] != a[i-1] && a[i] != a[i+1])
            {
                cout << i+1 << "\n";
                delete [] a;
                break;
            }
        }
    }
    return 0;
}

File: Codeforces/Kotlin/src/main/kotlin/problems/SpyDetected.kt
package problems.problems

/* Problem Link: https://codeforces.com/problemset/problem/1512/A */

fun main() {
    val tests = readln().toInt()

    for (i in 1..tests) {
        val n = readln().toInt()
        val numbers = readln().split(" ")

        if (numbers[0] != numbers[1] && numbers[0] != numbers[2]) {
            println(1)
            continue
        } else if (numbers[n - 1] != numbers[n - 2] && numbers[n - 1] != numbers[n - 3]) {
            println(n)
            continue
        } else {
            for (j in 1..<n - 1) {
                if (numbers[j] != numbers[j - 1] && numbers[j] != numbers[j + 1]) {
                    println(j + 1)
                    break
                }
            }
        }
    }
}